snforge_std/src/trace.cairo

```diff
@@ -0,0 +1,64 @@
+use starknet::testing::cheatcode;
+use starknet::ContractAddress;
+use super::PrintTrait;
+
+#[derive(Drop, Serde)]
+struct CallEntryPoint {
+    entry_point_type: EntryPointType,
+    entry_point_selector: felt252,
+    calldata: Array<felt252>,
+    storage_address: ContractAddress,
+    caller_address: ContractAddress,
+    call_type: CallType,
+    initial_gas: u64,
+}
+
+#[derive(Drop, Serde)]
+enum EntryPointType {
+    Constructor,
+    External,
+    L1Handler,
+}
+
+#[derive(Drop, Serde)]
+enum CallType {
+    Call,
+    Delegate,
+}
+
+fn get_last_transaction_trace() -> Array<CallEntryPoint> {
+    let mut output = cheatcode::<'get_last_transaction_trace'>(array![].span());
+    Serde::<Array<CallEntryPoint>>::deserialize(ref output).unwrap()
+}
+
+impl CallEntryPointPrintImpl of PrintTrait<CallEntryPoint> {
```

Comment:
This is a bit ugly when printing, but didn't want to create a separate cheatcode just for printing trace info